File: src/main/java/com/zunza/buythedip/crypto/exception/CryptoNotFoundException.java
package com.zunza.buythedip.crypto.exception;

import org.springframework.http.HttpStatus;

import com.zunza.buythedip.common.CustomException;

public class CryptoNotFoundException extends CustomException {
	private static final String MESSAGE = "암호화폐가 존재하지 않습니다.";

	public CryptoNotFoundException() {
		super(MESSAGE);
	}

	@Override
	public int getStatusCode() {
		return HttpStatus.NOT_FOUND.value();
	}
}

File: src/main/kotlin/com/zunza/buythedip/crypto/exception/CryptoNotFoundException.kt
package com.zunza.buythedip.crypto.exception

import com.zunza.buythedip.common.CustomException
import org.springframework.http.HttpStatus

class CryptoNotFoundException(
    private val cryptoId: Long
) :  CustomException(
    MESSAGE.format(cryptoId)
){
    companion object {
        private const val MESSAGE = "암호화폐를 찾을 수 없습니다 CRYPTO ID: %d"
    }

    override fun getStatusCode() = HttpStatus.NOT_FOUND.value()
}
